### PR TITLE
Use GetRequiredService instead of GetService to resolve jobs

### PR DESCRIPTION
**Pull request by skarum**
As discussed in #23 

Use GetRequiredService to resolve jobs.

Added tests + modified test that broke due to this change.

### CODE REVIEW

File: tests/NCronJob.Tests/NCronJobIntegrationTests.cs [hunk 1]
```diff
@@ -163,6 +165,36 @@ public async Task NotRegisteredJobShouldNotAbortOtherRuns()
...
+        Assert.Throws<InvalidOperationException>(() =>
+        {
+            using var executor = new JobExecutor(provider, NullLogger<JobExecutor>.Instance);
+            executor.RunJob(new RegistryEntry(typeof(JobWithDependency), new JobExecutionContext(null), null), CancellationToken.None);
+
+        });
+    }
+
+    [Fact]
+    public async Task NotThrowIfJobWithDependenciesRegistered()
```
Comment (linkdotnet) [inline tests/NCronJob.Tests/NCronJobIntegrationTests.cs:184]: That is basically tested by many other tests indirectly. I would remove that test.

**Review by linkdotnet — COMMENTED** [review]
LGTM! Thanks for providing a fix and a test.

Can you add an entry to https://github.com/linkdotnet/NCronJob/blob/main/CHANGELOG.md as I use this file for releases.

File: tests/NCronJob.Tests/NCronJobIntegrationTests.cs (hunk 1 above)
Comment (skarum) [reply tests/NCronJob.Tests/NCronJobIntegrationTests.cs]: Good point - it's removed

**Comment by skarum** [comment]
> Can you add an entry to https://github.com/linkdotnet/NCronJob/blob/main/CHANGELOG.md as I use this file for releases.

Done

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ All notable changes to **NCronJob** will be documented in this file. The project
 ### Changed
 
 - Implementation of the scheduling. Better performance and closed some memory leaks
+- Throw exception if job cannot be resolved with dependencies from the DI container
```
Comment (linkdotnet) [inline CHANGELOG.md:12]:
```suggestion
- Throw exception if job cannot be resolved with dependencies from the DI container. Reported and implemented by [@skarum](https://github.com/skarum) in [#23)(https://github.com/linkdotnet/NCronJob/issues/23)
```

**Comment by linkdotnet** [comment]
Perfect! Thanks again. I will make a release `2.0.5` after the merge!

**Pull request merged**